app/src/org/commcare/activities/PostRequestActivity.java

```diff
@@ -108,13 +112,8 @@ private void performSync() {
 
     private void makePostRequest() {
         if (!hasTaskLaunched && !inErrorState) {
-            ModernHttpTask postTask;
-            try {
-                postTask = new ModernHttpTask(this, url, params, true, null);
-            } catch (ModernHttpRequester.PlainTextPasswordException e) {
-                enterErrorState(Localization.get("post.not.using.https", url.toString()));
-                return;
-            }
+            RequestBody requestBody = ModernHttpRequester.getPostBody(params);
+            ModernHttpTask postTask = new ModernHttpTask(this, url.toString(), new HashMap(), requestBody, HTTPMethod.POST, null);
```

Comment:
Now I don't think it's a good solution as doing that will cause another overloaded constructor in `CommCareApplication` and yet another in `ModernHTTPTask` which doesn't seem to be worth for removing an argument. Another not so good approach would be to have a setter for `requestBody` in `ModernHttpRequester` instead of passing it as an argument to constructor. I am not sure making `requestBody`mutable is a good idea and hence of the opinion to keep it same as now.

Reply:
Hmm.. I would probably be slightly in favor of having a setter method for the request body, but understand your hesitation. Will leave it up to you.